Lib/ufo2ft/util.py

```diff
@@ -47,19 +50,44 @@ def makeOfficialGlyphOrder(font, glyphOrder=None):
 
 
 class _GlyphSet(dict):
-
     @classmethod
-    def from_layer(cls, font, layerName=None, copy=False):
+    def from_layer(cls, font, layerName=None, copy=False, skipExportGlyphs=None):
+        """Return a mapping of glyph names to glyph objects from `font`."""
         if layerName is not None:
             layer = font.layers[layerName]
         else:
             layer = font.layers.defaultLayer
+
         if copy:
             self = _copyLayer(layer, obj_type=cls)
             self.lib = deepcopy(layer.lib)
         else:
             self = cls((g.name, g) for g in layer)
             self.lib = layer.lib
+
+        # If any glyphs in the skipExportGlyphs list are used as components, decompose
+        # them in the containing glyphs...
+        if skipExportGlyphs:
+            for glyph in self.values():
+                if any(c.baseGlyph in skipExportGlyphs for c in glyph.components):
+                    deepCopyContours(self, glyph, glyph, Transform(), skipExportGlyphs)
+                    try:  # defcon: `glyph.components` returns a new list.
+                        glyph._components = [
```

Comment:
hm.. is there not a better way to update a glyph's components in defcon than accessing an a private attribute?

Reply:
There is no direct public access it seems: https://github.com/robotools/defcon/blob/master/Lib/defcon/objects/glyph.py#L678

I could use the `removeComponent` method just for defcon, though...

Reply:
that would be better